metagraph/src/graph/representation/masked_graph.hpp

```diff
@@ -115,8 +115,8 @@ class MaskedDeBruijnGraph : public DeBruijnGraph {
         kmers_in_graph_ = std::move(mask);
     }
 
-    typedef std::function<void(const std::function<void(node_index, bool)>&)> GenerateNodes;
-    virtual void update_mask(const GenerateNodes &generate_nodes,
+    typedef std::function<void(const std::function<void(node_index, bool /* new value */)>&)> GenerateNodeUpdates;
```

Comment:
"new value" means added or removed node?

Reply:
should I call it "update value" instead? It's the value that should be stored at that position

Reply:
Ah, I'm confused again. What does this function actually do?

Reply:
calls all nodes in the full graph and indicates those from the subgraph with 1/0?

Reply:
It generates a bunch of `(node_index, bool)` pairs. So for each `(i,j)` pair, it does `mask[i] = j`

Reply:
ah, I see. Why not pass an abstract `bitmap`?
You can define it from a callback, so the complexity will be equivalent, but much easier to read.

Reply:
I see now that bitmap wouldn't be sufficient. There is a design issue, this interface is not ideal (see the other comment in .cpp file)